torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -3382,16 +3389,75 @@ at::Tensor interpolate(
     }
   }
 
+  float scale_factors_1 = -1.0f;
+  float scale_factors_2 = -1.0f;
+  float scale_factors_3 = -1.0f;
+
+  if(!scale_factors.isNone() && use_scale_factor == c10::nullopt) {
+    use_scale_factor = false;
+    bool warn_use_scale_factor = false;
+
+    if (scale_factors.isDouble()) {
+      if(_is_float_value(static_cast<float>(scale_factors.toDouble()))){
+        warn_use_scale_factor = true;
+      }
+    } else if (scale_factors.isDoubleList()) {
+      auto scale_factors_list_ref = scale_factors.toDoubleListRef();
+      std::vector<double> scale_factors_vec(scale_factors_list_ref.begin(), scale_factors_list_ref.end());
+      auto scale_factors_list = c10::impl::toList(scale_factors_vec);
```

Comment:
`c10::List` api is kind of hard to use, but still we should be just constructing the `c10::List` here instead of constructing both a `std::vector<double>` and a `c10::List`

Reply:
I followed the same logic as convert_scale_factor_to_double to construct these lists